### DISCUSSION TITLE
How can i acheive transparency for the bg?

### DISCUSSION THREAD

**Discussion by semi710** [tiagovla/tokyodark.nvim#26] [Q&A] [answered]
like on Nvimtree, lualine, bufferline and all floating windows?

**Comment by tiagovla** [comment, marked as answer]
```lua
return {
    "tiagovla/tokyodark.nvim",
    opts = {
        custom_highlights = function(hl, p)
            for h, v in pairs(hl) do
                if v.bg then
                    hl[h].bg = p.none
                end
            end
            return {}
        end,
    },
    config = function(_, opts)
        require("tokyodark").setup(opts)
        require("tokyodark").colorscheme()
    end,
}
```

This should remove all backgrounds. Or you could modify it to replace just the ones you want. Check the `highlights.lua` file.

**Reply by semi710** [reply]
Everythign is working fine just the search highlight and the lazy highlight is blacked out.
![image](https://github.com/tiagovla/tokyodark.nvim/assets/60490474/20db7253-3383-483d-8e90-ba4a4ea7f7ce)
![image](https://github.com/tiagovla/tokyodark.nvim/assets/60490474/2151a3da-f1e6-4885-96a2-8efb5a9ebf85)

**Reply by tiagovla** [reply]
It might be a bit extreme to remove all. Instead, consider removing the background of `Normal`, `NormalFloat`, and any other background highlights used by your plugins.

For reference, here are the ones used by lazy.nvim (check the default group column): https://github.com/folke/lazy.nvim/blob/2a9354c7d2368d78cbd5575a51a2af5bd8a6ad01/doc/lazy.nvim.txt#L885-L950

**Reply by semi710** [reply]
was able to acheive what i wanted 
```lua
local groups = {
				"Normal",
				"NormalFloat",
				"NormalNC",
				"Comment",
				"Constant",
				"Special",
				"Identifier",
				"Statement",
				"PreProc",
				"Type",
				"Underlined",
				"Todo",
				"String",
				"Function",
				"Conditional",
				"Repeat",
				"Operator",
				"Structure",
				"LineNr",
				"NonText",
				"SignColumn",
				"CursorLineNr",
				"EndOfBuffer",
			}
			hl["Normal"].bg = palette.none
			for _, group in ipairs(groups) do
				if hl[group] ~= nil then
					hl[group].bg = palette.none
				end
			end```
Thanks buddy
